patches/minecraft/net/minecraft/entity/EntityLivingBase.java.patch

```diff
@@ -1,22 +1,38 @@
 --- ../src-base/minecraft/net/minecraft/entity/EntityLivingBase.java
 +++ ../src-work/minecraft/net/minecraft/entity/EntityLivingBase.java
-@@ -82,6 +82,7 @@
+@@ -82,11 +82,14 @@
      private static final Logger field_190632_a = LogManager.getLogger();
      private static final UUID field_110156_b = UUID.fromString("662A6B8D-DA3E-4C1C-8813-96EA6097278D");
      private static final AttributeModifier field_110157_c = (new AttributeModifier(field_110156_b, "Sprinting speed boost", 0.30000001192092896D, 2)).func_111168_a(false);
 +    public static final net.minecraft.entity.ai.attributes.IAttribute SWIM_SPEED = new net.minecraft.entity.ai.attributes.RangedAttribute(null, "forge.swimSpeed", 1.0D, 0.0D, 1024.0D).func_111112_a(true);
      protected static final DataParameter<Byte> field_184621_as = EntityDataManager.<Byte>func_187226_a(EntityLivingBase.class, DataSerializers.field_187191_a);
      private static final DataParameter<Float> field_184632_c = EntityDataManager.<Float>func_187226_a(EntityLivingBase.class, DataSerializers.field_187193_c);
      private static final DataParameter<Integer> field_184633_f = EntityDataManager.<Integer>func_187226_a(EntityLivingBase.class, DataSerializers.field_187192_b);
-@@ -188,6 +189,7 @@
+     private static final DataParameter<Boolean> field_184634_g = EntityDataManager.<Boolean>func_187226_a(EntityLivingBase.class, DataSerializers.field_187198_h);
+     private static final DataParameter<Integer> field_184635_h = EntityDataManager.<Integer>func_187226_a(EntityLivingBase.class, DataSerializers.field_187192_b);
++    private static final DataParameter<Float> LAST_HEALTH = EntityDataManager.func_187226_a(EntityLivingBase.class, DataSerializers.field_187193_c);
++    private static final DataParameter<NBTTagCompound> WATCHER_POTION_EFFECTS = EntityDataManager.func_187226_a(EntityLivingBase.class, DataSerializers.field_192734_n);
```

Comment:
It'd be great if you could make this an `Optional<NBTTagCompound>` instead, and save creating a compound tag for *every single living entity that is constructed*.

Reply:
I am not familiar with the Optional class, but I assume you are talking about the com.google.common.base.Optional class? So I don't quite understand your suggestion or how I would go about implementing it. I'll try playing around with it for a bit to see if I can grasp it.

Reply:
I also use DataParameter because I rely on its ability to automatically send and receive the relevant packets to keep the value consistent across server and client. Can Optional do that?

Reply:
DataParamaeters are not allows as they break network compatibility with vanilla and older clients.

Reply:
@LexManos 
What does that mean? Are you saying my patch is invalid since it adds 2 new data parameters?